usaspending_api/recipient/management/sql/recipient_lookup/030_sam_parents.sql

```diff
@@ -10,7 +10,7 @@ INSERT INTO public.temporary_restock_recipient_lookup (
 )
 SELECT
   DISTINCT ON (ultimate_parent_unique_ide, ultimate_parent_legal_enti)
```

Comment:
Can you add a bugfix here while you are changing things:
```
      -- BUG fixed here: 
      --   - parent DUNS ID (value of ultimate_parent_unique_ide) can show up with the same value on many rows
      --   - on those rows, it could have different (i.e. older) names for itself
      --   - originally this was partitioning on BOTH the parent DUNS and parent name. 
      --   - That may introduce the same DUNS multiple times with different names, which would yield entering the same recipient_hash value into the destination table more than once.
      --   - FIXED: by partitioning only on the parent DUNS. It will take the parent Name from the most recent row
```

i.e. it should be
```
SELECT
  DISTINCT ON (ultimate_parent_unique_ide)
```

Note that this was discovered when prototyping this ETL using `MERGE` and `PARTITION BY`. Replace "partition by" above with "distinct on".